src/components/Player/Pages/Histograms/Histograms.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Graph } from 'components/Visualizations';
+import { getPlayerHistogram } from 'actions';
+import { playerHistogram } from 'reducers';
+import { withRouter } from 'react-router';
+import FlatButton from 'material-ui/FlatButton';
+import { deSnake } from 'utility';
+import histogramNames from './histogramNames';
+import styles from './Histograms.css';
+
+
+const getAxis = (name, show = true) => ({
+  label: {
+    text: name,
+    show,
+  },
+});
+
+const selectHistogram = (router, histogramName, playerId) => {
+  router.push(`/players/${playerId}/histograms/${histogramName}`);
+};
+
+const Histogram = ({ histogramName, columns, router, playerId }) => (
+  <div style={{ fontSize: 10 }}>
+    <div className={styles.buttonContainer}>
+      {histogramNames.map((histogram, index) => (
+        <FlatButton
+          onClick={() => selectHistogram(router, histogram, playerId)}
+          className={histogramName === histogram ? styles.selectedButton : styles.button}
+          key={index}
+        >
+          <span className={styles.buttonText}>{deSnake(histogram)}</span>
+        </FlatButton>
+      ))}
+    </div>
+    <Graph
+      columns={columns}
+      name={histogramName}
+      type="bar"
+      xAxis={getAxis(histogramName)}
+      yAxis={getAxis('Matches')}
+    />
+  </div>
+);
+
+const getData = props => {
+  props.getPlayerHistogram(props.playerId, props.histogramName);
+};
+
+class RequestLayer extends React.Component {
+  componentWillMount() {
+    if (!this.props.histogramName) {
+      this.props.router.push(`/players/${this.props.playerId}/histograms/${histogramNames[0]}`);
+    } else {
+      getData(this.props);
+    }
+  }
+
+  componentWillUpdate(nextProps) {
+    if (!nextProps.histogramName) {
+      this.props.router.push(`/players/${this.props.playerId}/histograms/${histogramNames[0]}`);
```

Comment:
This is checking if when the component is updating, does it have a histogram name which is truthy as a prop. So if the user goes to /histograms, they are redirected to /histograms/kills. This has the issue mentioned in this [comment](https://github.com/odota/ui/pull/188/files/a4a36aa09aa20365261176d76b043ecf14fd2935#diff-c912dcb697b27f45c8c4cf4f06cc32f1), but I think it's acceptable because it's an edge case (vs when people click on the histogram tab, which is why I refactored that to be different).